File: frontend/packages/@depmap/types/src/Group.ts
import { Dataset } from "./Dataset";

export default interface Group {
  id: string;
  name: string;
  group_entries: GroupEntry[];
  datasets: Dataset[];
}

export interface GroupEntry {
  access_type: AccessType;
  email: string;
  exact_match: boolean;
  id: string;
}

export interface GroupArgs {
  name: string;
}

export enum AccessType {
  read = "read",
  write = "write",
  owner = "owner",
}

export interface GroupEntryArgs {
  email: string;
  exact_match: boolean;
  access_type: AccessType;
}

export interface GroupTableData {
  id: string;
  name: string;
  groupEntries: GroupEntry[];
  groupEntriesCount: number;
  datasetsCount: string;
}

File: breadbox/breadbox/schemas/group.py
from pydantic import BaseModel

import enum
from .common import DBBase


class AccessType(enum.Enum):
    read = "read"
    write = "write"
    owner = "owner"


class GroupEntryBase(BaseModel):
    email: str
    exact_match: bool = True
    access_type: AccessType


class GroupEntryIn(GroupEntryBase):
    pass


class GroupEntry(GroupEntryBase, DBBase):
    pass


class GroupBase(BaseModel):
    name: str


class GroupIn(GroupBase):
    pass


class Group(GroupBase, DBBase):
    pass
